osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySliderBall.cs

```diff
@@ -79,22 +85,19 @@ protected override void UpdateAfterChildren()
 
         private void updateStateTransforms(DrawableHitObject obj, ArmedState _)
         {
-            if (obj is not DrawableSlider)
-                return;
-
-            using (BeginAbsoluteSequence(slider.StateUpdateTime))
+            using (BeginAbsoluteSequence(drawableObject.AsNonNull().StateUpdateTime))
```

Comment:
Rather than `AsNonNull()`, you can add a `Debug.Assert(drawableObject != null)` to the start of this method.